### PR TITLE
bugfix: Init & setMode should be backward-compatible

### PR DESCRIPTION
**Pull request by brikendr**
Cannot call setMode on a camera running a firmware where this feature is
not supported (was introduced with the Plaza support). The fix involves
making the options "mode" to defaul to 'NORMAL' which means the camera
is in autozoom (GF) mode by default.

Adding additional tests for testing all the edge cases when updating the
autozoom control options.

### CODE REVIEW

File: src/components/autozoomControl.ts [hunk 1]
```diff
@@ -93,11 +97,15 @@ export default class AutozoomControl implements ICnnControl {
   }
 
   private _validateOptions(options: AutozoomControlOpts) {
-    if (
-      options.shouldAutoFrame === false &&
-      options.mode &&
-      options.mode !== AutozoomModes.NORMAL
-    ) {
+    if (!options.mode || !Object.values(AutozoomModes).includes(options.mode)) {
```
Comment (eskilgh) [inline src/components/autozoomControl.ts:100]: Do we need to check `!options.mode`, isn't it enough to just do the includes check?

File: src/components/autozoomControl.ts [hunk 2]
```diff
@@ -93,11 +97,15 @@ export default class AutozoomControl implements ICnnControl {
...
+      throw Error(`The following mode is not supported on autozoom controller: ${options.mode}`);
+    }
+
+    if (options.shouldAutoFrame === undefined || options.shouldAutoFrame === null) {
```
Comment (eskilgh) [inline src/components/autozoomControl.ts:104]: Maybe do `if (!typeof(options.shouldAutoFrame) === "boolean")` instead?

File: src/components/autozoomControl.ts [hunk 3]
```diff
@@ -93,11 +97,15 @@ export default class AutozoomControl implements ICnnControl {
...
+      throw Error(`Option 'shouldAutoFrame' cannot not be set to ${options.shouldAutoFrame}`);
+    }
+
+    if (options.shouldAutoFrame === false && options.mode !== AutozoomModes.NORMAL) {
```
Comment (eskilgh) [inline src/components/autozoomControl.ts:108]: `options.shouldAutoFrame === false` .. :)

File: tests/components/autozoomControl.spec.ts [hunk 4]
```diff
@@ -18,6 +21,53 @@ describe('AutozoomControl', () => {
     autozoomControl = new AutozoomControl(deviceManager);
   });
 
+  describe('Constructor', () => {
```
Comment (eskilgh) [inline tests/components/autozoomControl.spec.ts:24]: nice

**Review by eskilgh — CHANGES_REQUESTED** [review]
Thanks for fixing this.

One thing to be weary of is that we are now pretty strict when setting the options.mode, so it has to be one of the modes from the enum. However - as I've understood from Jannes there are more modes available that could be relevant in some debugging scenarios. Probably not a big deal right now, but it could perhaps be appropriate to relax the check a bit to just check for a number or something

File: src/components/autozoomControl.ts (hunk 1 above)
Comment (brikendr) [reply src/components/autozoomControl.ts:100]: Good point

File: src/components/autozoomControl.ts (hunk 3 above)
Comment (brikendr) [reply src/components/autozoomControl.ts:108]: Not sure I understand your suggestion here!

Comment (brikendr) [reply src/components/autozoomControl.ts]: Got it, thanks for your review Eskil :D

**Review by eskilgh — APPROVED** [review]
🥇
